backend/src/main/java/ai/verta/modeldb/cron_jobs/ParentTimestampUpdateCron.java

```diff
@@ -0,0 +1,113 @@
+package ai.verta.modeldb.cron_jobs;
+
+import ai.verta.modeldb.utils.ModelDBHibernateUtil;
+import java.util.TimerTask;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+public class ParentTimestampUpdateCron extends TimerTask {
+  private static final Logger LOGGER = LogManager.getLogger(ParentTimestampUpdateCron.class);
+
+  public ParentTimestampUpdateCron() {}
+
+  /** The action to be performed by this timer task. */
+  @Override
+  public void run() {
+    LOGGER.info("ParentTimestampUpdateCron wakeup");
+
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      session.beginTransaction();
+      // Update project timestamp
+      updateProjectByExperimentTimestamp(session);
+
+      // Update experiment timestamp
+      updateExperimentByExperimentRunTimestamp(session);
```

Comment:
Reverse the order. First update the lower elements of the hierarchy, then the top ones.

Reply:
Fixed